### ISSUE TITLE
ssm eats ^S & ^O chars? so emacs (save) and screen (detach) are problematic

### ISSUE THREAD

**Issue by mvgfr** [aws/amazon-ssm-agent#174]
expected behavior, if I ssh directly to an instance instead of using ssm

**Comment by sahirangaraju** [comment]
Thank you for the feedback! We will look into this.

**Comment by linuxbaby03** [comment]
+1 Please address this!!! thank you kindly

**Comment by keith-wellman** [comment]
+1 to address.

(To be fair - unclear to me if it is the ssm-agent that receives then swallows the ctrl codes, or whether this might be on the cli session manager component failing to pass them through.  But I'll start here)

**Comment by bogbert** [comment]
+1
As a constant emacs user, this issue drives me mad. A fix would be much appreciated.

**Comment by mvgfr** [comment]
> Thank you for the feedback! We will look into this.

any news?

**Comment by bogbert** [comment]
> > Thank you for the feedback! We will look into this.
> 
> any news?

- This problem is related to the client, not the agent, so I guess this ticket should have been opened here: https://github.com/aws/session-manager-plugin/issues
- The workaround I'm using is to run the command `stty -ixon` before I initiate the ssm session. This command is used to disable the XON/XOFF control flow so that the terminal doesn't capture Ctrl+S and Ctrl+Q. I suppose the ssh client is capable of doing something similar by itself, and session-manager-plugin should probably follow its example. Once you're done with ssm, if you want to put your terminal back to its initial state, run `stty ixon`

**Comment by mvgfr** [comment]
> * The workaround I'm using is to run the command `stty -ixon`…

No; that's a different issue - and most of us (including me) already do that -- which does not prevent _this_ problem.

**Comment by yaitskov** [comment]
Common..

**Comment by dacut** [comment]
> * This problem is related to the client, not the agent, so I guess this ticket should have been opened here: https://github.com/aws/session-manager-plugin/issues

@bogbert, how did you deduce this? Just curious; this has been on my radar to "get to if nobody else does first" for awhile now, and trying to figure out where in the flow (client vs agent) was going to be the first step. Thanks!

**Comment by bogbert** [comment]
> @bogbert, how did you deduce this? Just curious; this has been on my radar to "get to if nobody else does first" for awhile now, and trying to figure out where in the flow (client vs agent) was going to be the first step. Thanks!

I did not experience the issue when opening an ssm session from the AWS console, which seemed to point in the direction of a client issue. This was then confirmed when it turned out my problem was caused the terminal XON/XOFF control flow. But apparently this thread is actually about a different defect (see the response by mvgfr), so I guess my post was off topic.

**Comment by dacut** [comment]
@bogbert, Ah, gotcha. Thanks for the update!
